phoenix-core/src/test/java/org/apache/phoenix/index/IndexScrutinyMapperTest.java

```diff
@@ -0,0 +1,117 @@
+        this.isNamespaceEnabled = isNamespaceEnabled;
+    }
+    @Test
+    public void testGetSourceTableName_table() {
+        String fullTableName = SchemaUtil.getQualifiedTableName(schema, tableName);
+        PName sourcePhysicalName = SchemaUtil.getPhysicalHBaseTableName(schema, tableName,
+                isNamespaceEnabled);
+        String expectedName = SchemaUtil.getPhysicalTableName(Bytes.toBytes(fullTableName),
+                isNamespaceEnabled).toString();
+        //setup
+        Mockito.when(inputTable.getType()).thenReturn(PTableType.TABLE);
+        Mockito.when(inputTable.getPhysicalName()).thenReturn(sourcePhysicalName);
+        Mockito.when(inputTable.getTableName()).thenReturn(PNameFactory.newName(tableName));
+        Mockito.when(inputTable.getSchemaName()).thenReturn(PNameFactory.newName(schema));
+        //test
+        String output = IndexScrutinyMapper.getSourceTableName(inputTable, isNamespaceEnabled);
+        //assert
+        Assert.assertEquals(expectedName, output);
+    }
+
+    @Test
+    public void testGetSourceTableName_view() {
+        String fullTableName = SchemaUtil.getQualifiedTableName(schema, tableName);
+        PName sourcePhysicalName = SchemaUtil.getPhysicalHBaseTableName(schema, tableName,
+                isNamespaceEnabled);
+        String expectedName = SchemaUtil.getPhysicalTableName(Bytes.toBytes(fullTableName),
+                isNamespaceEnabled).toString();
+        //setup
+        Mockito.when(inputTable.getType()).thenReturn(PTableType.VIEW);
+        Mockito.when(inputTable.getPhysicalName()).thenReturn(sourcePhysicalName);
+        //test
+        String output = IndexScrutinyMapper.getSourceTableName(inputTable, isNamespaceEnabled);
+        //assert
+        Assert.assertEquals(expectedName, output);
+    }
+
+    @Test
+    public void testGetSourceTableName_index() {
+        String fullTableName = SchemaUtil.getQualifiedTableName(schema, indexName);
+        PName sourcePhysicalName = SchemaUtil.getPhysicalHBaseTableName(schema, indexName,
+                isNamespaceEnabled);
+        String expectedName = SchemaUtil.getPhysicalTableName(Bytes.toBytes(fullTableName),
+                isNamespaceEnabled).toString();
+
+        //setup
+        Mockito.when(inputTable.getType()).thenReturn(PTableType.INDEX);
+        Mockito.when(inputTable.getPhysicalName()).thenReturn(sourcePhysicalName);
+        Mockito.when(inputTable.getTableName()).thenReturn(PNameFactory.newName(indexName));
+        Mockito.when(inputTable.getSchemaName()).thenReturn(PNameFactory.newName(schema));
+
+        //test
+        String output = IndexScrutinyMapper.getSourceTableName(inputTable, isNamespaceEnabled);
+        //assert
+        Assert.assertEquals(expectedName, output);
+    }
+
+    @Test
+    public void testGetSourceTableName_viewIndex() {
+        PName physicalTableName = SchemaUtil.getPhysicalHBaseTableName(schema, tableName,
```

Comment:
I would suggest creating actual views, indexes, etc rather than mocking so that we are sure there is no test bug and the name convention is indeed as we expect.

Reply:
This is a unit test. Creating a view/table would involve setting up and tearing down mini-cluster which will lead to more time in build phase.